purchase_order_price_recalculation/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    show_update_line_info = fields.Boolean()
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        if (
+            self.partner_id
+            and self._origin.partner_id
+            and self._origin.partner_id != self.partner_id
+        ):
+            self.show_update_line_info = True
+        else:
+            self.show_update_line_info = False
```

Comment:
This is anyways not covering all cases. Maybe it's better to simply mark as True on the onchange.

```suggestion
    def _onchange_partner_id(self):
        self.show_update_line_info = (
            self.partner_id
            and self._origin.partner_id
            and self._origin.partner_id != self.partner_id
        )
```